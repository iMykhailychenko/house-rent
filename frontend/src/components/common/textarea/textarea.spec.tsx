import { fireEvent, render } from '@testing-library/react';

import Textarea from './textarea';

describe('Textarea component', () => {
    it('matches snapshot', () => {
        const handleChange = jest.fn();
        const { asFragment } = render(
            <Textarea
                value="test"
                onChange={handleChange}
                id="test"
                className="test"
                name="test"
                placeholder="test"
                error="error"
            />,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('password', () => {
        const handleChange = jest.fn();
        const { queryByText } = render(<Textarea value="test" onChange={handleChange} />);

        expect(queryByText('label')).toBeFalsy();
        expect(queryByText('error')).toBeFalsy();
    });

    it('error', () => {
        const handleChange = jest.fn();
        const { queryByText } = render(<Textarea value="test" onChange={handleChange} error="error" />);

        expect(queryByText('label')).toBeFalsy();
        expect(queryByText('error')).toBeInTheDocument();
    });

    it('label', () => {
        const handleChange = jest.fn();
        const { queryByText } = render(<Textarea value="test" onChange={handleChange} label="label" />);

        expect(queryByText('error')).toBeFalsy();
        expect(queryByText('label')).toBeTruthy();
    });

    it('events', () => {
        const handleChange = jest.fn();
        const handleBlur = jest.fn();
        const handleFocus = jest.fn();
        const handleKeyDown = jest.fn();
        const { getByPlaceholderText } = render(
            <Textarea
                value="test"
                placeholder="input"
                onChange={handleChange}
                onBlur={handleBlur}
                onFocus={handleFocus}
                onKeyDown={handleKeyDown}
            />,
        );

        fireEvent.focus(getByPlaceholderText('input'));
        expect(handleFocus).toHaveBeenCalledTimes(1);

        fireEvent.blur(getByPlaceholderText('input'));
        expect(handleBlur).toHaveBeenCalledTimes(1);

        fireEvent.change(getByPlaceholderText('input'), { target: { value: 'abc' } });
        expect(handleChange).toHaveBeenCalledTimes(1);

        fireEvent.keyDown(getByPlaceholderText('input'));
        expect(handleKeyDown).toHaveBeenCalledTimes(1);
    });
});
