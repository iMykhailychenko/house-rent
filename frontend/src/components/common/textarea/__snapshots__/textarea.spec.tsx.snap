// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="inner"
    >
      <div
        class="wrp"
      >
        <div
          class="tooltip bottom_right tooltip"
        >
          error
        </div>
        <textarea
          class="textarea test error"
          id="test"
          name="test"
          placeholder="test"
          wrap="soft"
        >
          test
        </textarea>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
